pandas/io/common.py

```diff
@@ -121,7 +124,7 @@ def _validate_header_arg(header):
         )
 
 
-def _stringify_path(filepath_or_buffer):
+def _stringify_path(filepath_or_buffer: FilePathOrBuffer) -> FilePathOrBuffer:
```

Comment:
Hmm surprised this doesn't raise but the return here isn't totally generic - doesn't this return a string? 

`FilePathOrBuffer` as a parameter and a return type should mean that there is no transformation of the type within the function body (i.e. if a pathlib.Path object comes in, you get a pathlib.Path object in return and not a str)

Reply:
> Hmm surprised this doesn't raise but the return here isn't totally generic - doesn't this return a string?

_stringify_path attempts to convert a path-like object to a string and returns maybe a string version of the object.

Any other object is passed through unchanged, which includes bytes, strings, buffers, or anything else that's not even path-like.

> `FilePathOrBuffer` as a parameter and a return type should mean that there is no transformation of the type within the function body (i.e. if a pathlib.Path object comes in, you get a pathlib.Path object in return and not a str)

FilePathOrBuffer is a Union not a TypeVar.

Reply:
Ah that's right - makes sense